### PR TITLE
Add foundation.e.easy-installer

### PR DESCRIPTION
**Pull request by shuriken1812**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://gitlab.e.foundation/e/devices/easy-installer
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

I am the qa team member at e foundation/ murena and worked on the flatpak build, this easy installer is so that people can easily install /e/ OS on their phones using the supported devices listed here https://doc.e.foundation/easy-installer
Once this is available on flathub we will update docs to include the flatpak instructions.

Additional link(s) ->

- issue to provide a flatpak build.
https://gitlab.e.foundation/e/backlog/-/issues/5545#note_240596

- About easy-installer 
https://doc.e.foundation/easy-installer

- Snap build
https://snapcraft.io/easy-installer

- Application ID 
https://gitlab.e.foundation/e/devices/easy-installer/-/issues/425

### CODE REVIEW

File: foundation.e.easy-installer.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+runtime-version: 21.08
+sdk: org.freedesktop.Sdk
+command: easy-installer
+
+finish-args:
+  - "--env=SNAP_NAME=easy-installer"
+  - "--env=SNAP_USER_COMMON=/var/tmp"
+  - "--share=network"
+  - "--share=ipc"
+  - "--socket=x11"
```
Comment (hfiguiere) [inline foundation.e.easy-installer.yml:12]: if it actually supports wayland

```suggestion
  - "--socket=fallback-x11"
```

File: foundation.e.easy-installer.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+app-id: foundation.e.easy-installer
+runtime: org.freedesktop.Platform
+runtime-version: 21.08
+sdk: org.freedesktop.Sdk
+command: easy-installer
+
+finish-args:
+  - "--env=SNAP_NAME=easy-installer"
+  - "--env=SNAP_USER_COMMON=/var/tmp"
```
Comment (hfiguiere) [inline foundation.e.easy-installer.yml:9]: SNAP? This is Flatpak.

File: foundation.e.easy-installer.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+modules:
+  - name: easy-installer
+    buildsystem: simple
+    build-commands:
+      - mv * /app
+      - install -D --mode=0644 /app/foundation.e.easy-installer.png /app/share/icons/hicolor/scalable/apps/foundation.e.easy-installer.png
+      - install -D --mode=0644 --target-directory=/app/share/applications/ /app/foundation.e.easy-installer.desktop
+    sources:
+      - type: archive
+        url: https://gitlab.e.foundation/e/devices/easy-installer/-/package_files/18678/download
```
Comment (hfiguiere) [inline foundation.e.easy-installer.yml:24]: this contain x86_64 binaries. Yet there is still an `aarch64` flatpak built.

**Review by hfiguiere — COMMENTED** [review]
There is no appstream file.

Both the icon and the .desktop file should come directly from upstream.

File: foundation.e.easy-installer.yml (hunk 2 above)
Comment (shuriken1812) [reply foundation.e.easy-installer.yml:9]: We use env vars SNAP_NAME and SNAP_USER_COMMON in the upstream project 
https://gitlab.e.foundation/e/devices/easy-installer/-/blob/master/src/main/java/ecorp/easy/installer/AppConstants.java#L220
will see so we can change it.

Comment (hfiguiere) [reply foundation.e.easy-installer.yml:9]: You should follow XDG standards insteads.

https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

Comment (hfiguiere) [reply foundation.e.easy-installer.yml:9]: also detecting if you are inside flatpak at runtime is done by checking for `/.flatpak-info`

**Comment by hfiguiere** [comment]
>     * Application ID
>       https://gitlab.e.foundation/e/devices/easy-installer/-/issues/425

This is 404.

File: foundation.e.easy-installer.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  - "--share=ipc"
+  - "--socket=x11"
+  - "--socket=wayland"
+  - "--device=all"
+modules:
+  - name: easy-installer
+    buildsystem: simple
+    build-commands:
+      - mv * /app
+      - install -D --mode=0644 /app/foundation.e.easy-installer.png /app/share/icons/hicolor/scalable/apps/foundation.e.easy-installer.png
```
Comment (hfiguiere) [inline foundation.e.easy-installer.yml:20]: a PNG is NOT a scalable icon.

Comment (hfiguiere) [reply foundation.e.easy-installer.yml:20]: also it's not square and not any size that is standard. You need at least a 128x128 px.

**Pull request closed without merging**